include/SDL3/SDL_gpu.h

```diff
@@ -24,7 +24,118 @@
+ * Render Passes can render to up to four color textures and one depth texture simultaneously.
+ * If the set of textures being rendered to needs to change, the Render Pass must be ended and a new one must be begun.
+ *
+ * The app calls SDL_BeginGPURenderPass().
+ * Then it sets states it needs for each draw:
+ *
+ * - SDL_BindGPUGraphicsPipeline
+ * - SDL_SetGPUViewport
+ * - SDL_BindGPUVertexBuffers
+ * - SDL_BindGPUVertexSamplers
+ * - etc
+ *
+ * Then, make the actual draw commands with these states:
+ *
+ * - SDL_DrawGPUPrimitives
+ * - SDL_DrawGPUPrimitivesIndirect
+ * - SDL_DrawGPUIndexedPrimitivesIndirect
+ * - etc
+ *
+ * After all the drawing commands for a pass are complete, the app should call
+ * SDL_EndGPURenderPass(). Once a render pass ends all render-related state is reset.
+ *
+ * The app can begin new Render Passes and make new draws in the same command buffer
+ * until the entire scene is rendered.
+ *
+ * Once all of the render commands for the scene are complete,
+ * the app calls SDL_SubmitGPUCommandBuffer() to send it to the GPU for processing.
+ *
+ * If the app needs to read back data from texture or buffers, the API
+ * has an efficient way of doing this, provided that the app is willing to tolerate some latency.
+ * When the app uses SDL_DownloadFromGPUTexture() or SDL_DownloadFromGPUBuffer(), submitting the command buffer with
+ * SubmitGPUCommandBufferAndAcquireFence() will return a fence handle that the app
+ * can poll or wait on in a thread. Once the fence indicates that the command buffer is done processing,
+ * it is safe to read the downloaded data. Make sure to call SDL_ReleaseGPUFence() when done with the fence.
+ *
+ * The API also has "compute" support. The app calls SDL_GPUBeginComputePass()
+ * with compute-writeable textures and/or buffers, which can be written to in a compute shader.
+ * Then it sets states it needs for the compute dispatches:
+ *
+ * - SDL_BindGPUComputePipeline
+ * - SDL_BindGPUComputeStorageBuffers
+ * - SDL_BindGPUComputeStorageTextures
+ *
+ * Then, dispatch compute work:
+ *
+ * - SDL_DispatchGPUCompute
+ *
+ * For advanced users, this opens up powerful GPU-driven workflows.
+ *
+ * Graphics and compute pipelines require the use of shaders, which as mentioned above are small programs
+ * executed on the GPU. Each backend (Vulkan, Metal, D3D12) requires a different shader format.
+ * When the app creates the GPU device, the app lets the device know which shader formats the app can provide.
+ * It will then select the appropriate backend depending on the available shader formats and the backends available on the platform.
+ * When creating shaders, the app must provide the correct shader for the selected backend.
+ * If you would like to learn more about why the API works this way, there is a
+ * detailed [blog post](https://moonside.games/posts/layers-all-the-way-down/)
+ * explaining this situation. In the future, SDL may provide a universal shader format.
```

Comment:
```suggestion
 * explaining this situation.
```

Let's not make any promises here. It's very possible, but I don't want people to hold their breath. :)

Reply:
Good call.